source/functions/SimulinkModelFunctions/irregnLYaw.m

```diff
@@ -0,0 +1,89 @@
+function [Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = irregnLYaw(A,w,dofGRD,dirGRD,wGRD,fEHRE,fEHIM, fEHMD, phaseRand,dw,time,WaveDir,WaveSpread, Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM)
+%#codegen
+% Fext is the excitation force output, relYawLast is the angle of relative
+% yaw at which that last interpolation was performed. If the current
+% relative yaw angle - last interpolated > threshold, the interpolation is
+% performed again. To interpolate every time step, let threshold=0.
+
```

Comment:
The Simulink function block `IrregnLYaw` has some of this code duplicated, was that intended?

``` matlab

function [Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = irregnLYaw(A,w,dofGRD,dirGRD,wGRD,fEHRE,fEHIM, fEHMD, phaseRand,dw,time,WaveDir,WaveSpread, Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM)
%#codegen
% Fext is the excitation force output, relYawLast is the angle of relative
% yaw at which that last interpolation was performed. If the current
% relative yaw angle - last interpolated > threshold, the interpolation is
% performed again. To interpolate every time step, let threshold=0.

% caseFlag is for debug, delete when satisfied
A1=bsxfun(@plus,w*time,pi/2);
%initialize outputs
Fext = zeros(1,6);
% Fext = zeros(1,size(fExtRE,3));
relYawLast=0;
coeffsLastMD=zeros(length(w),6);
coeffsLastRE=zeros(length(w),6);
coeffsLastIM=zeros(length(w),6);
% coeffsLastMD=zeros(length(w),size(fExtRE,3));
% coeffsLastRE=zeros(length(w),size(fExtRE,3));
% coeffsLastIM=zeros(length(w),size(fExtRE,3));

[Fext,relYawLast,coeffsLastMD,coeffsLastRE,coeffsLastIM] = irregnLYaw(A,w,dofGRD,dirGRD,wGRD,fEHRE,fEHIM, fEHMD, phaseRand,dw,time,WaveDir,WaveSpread, Disp, intThresh, prevYaw, prevCoeffMD, prevCoeffRE, prevCoeffIM);

end

```

Reply:
The irregnLYaw case has to be initialized both within the matlab function from simulink and the m-file to avoid errors. Not exactly sure of the reason for the double initialization, but may be due to the higher dimensions.

Reply:
@jtgrasb thanks for looking into this. I [added this ](https://github.com/WEC-Sim/WEC-Sim/projects/58#card-66339777 )to the project board as something to debug later.